src/Libraries/Microsoft.Extensions.DataIngestion/Chunkers/DocumentTokenChunker.cs

```diff
@@ -0,0 +1,87 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Runtime.CompilerServices;
+using System.Text;
+using System.Threading;
+using Microsoft.ML.Tokenizers;
+using Microsoft.Shared.Diagnostics;
+
+namespace Microsoft.Extensions.DataIngestion.Chunkers
+{
+    /// <summary>
+    /// Processes a document by tokenizing its content and dividing it into overlapping chunks of tokens.
+    /// </summary>
+    /// <remarks>This class uses a tokenizer to convert the document's content into tokens and then splits the
+    /// tokens into chunks of a specified size, with a configurable overlap between consecutive chunks. The resulting
+    /// chunks are returned as a list of <see cref="IngestionChunk{T}"/> objects.</remarks>
+    public sealed class DocumentTokenChunker : IngestionChunker<string>
+    {
+        private static string GetDocumentMarkdown(IngestionDocument document)
+        {
+            StringBuilder sb = new();
+            for (int i = 0; i < document.Sections.Count; i++)
+            {
+                _ = sb.Append(document.Sections[i].GetMarkdown());
+                if (i != document.Sections.Count - 1)
+                {
+                    _ = sb.AppendLine();
+                }
+            }
+            return sb.ToString();
+        }
+
+        private readonly Tokenizer _tokenizer;
+        private readonly int _maxTokensPerChunk;
+        private readonly int _chunkOverlap;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="DocumentTokenChunker"/> class.
+        /// </summary>
+        /// <param name="options">The options for the chunker.</param>
+        public DocumentTokenChunker(IngestionChunkerOptions options)
+        {
+            _ = Throw.IfNull(options);
+
+            _tokenizer = options.Tokenizer;
+            _maxTokensPerChunk = options.MaxTokensPerChunk;
+            _chunkOverlap = options.OverlapTokens;
+        }
+
+        /// <inheritdoc/>
+        public override async IAsyncEnumerable<IngestionChunk<string>> ProcessAsync(IngestionDocument document, [EnumeratorCancellation] CancellationToken cancellationToken = default)
+        {
+            _ = Throw.IfNull(document);
+
+            string documentMarkdown = GetDocumentMarkdown(document);
+            int[] tokens = _tokenizer.EncodeToIds(documentMarkdown).ToArray();
```

Comment:
The `considerNormalization` is passed as `true` for this method. Should that be the default? Or should we disable the normalization?